pkg/blockdeviceclaim/v1alpha1/build.go

```diff
@@ -304,3 +307,31 @@ func (b *Builder) WithBlockDeviceTag(bdTagValue string) *Builder {
 	b.BDC.Object.Spec.Selector.MatchLabels[bdTagKey] = bdTagValue
 	return b
 }
+
+// WithSelector appends (or creates) the BDC Label Selector
+// by setting the provided key and corresponding value
+// This will enable the NDM to pick only devices that
+// match the given label and its value.
+func (b *Builder) WithSelector(labelSelector map[string]string) *Builder {
+	if len(labelSelector) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build BDC object: missing label selector fields"),
+		)
+		return b
+	}
+
+	if b.BDC.Object.Spec.Selector == nil {
+		b.BDC.Object.Spec.Selector = &metav1.LabelSelector{}
+	}
+	if b.BDC.Object.Spec.Selector.MatchLabels == nil {
+		b.BDC.Object.Spec.Selector.MatchLabels = map[string]string{}
+	}
+
+	for key, value := range labelSelector {
+		if len(strings.TrimSpace(value)) != 0 {
```

Comment:
Why is this check required here?

Reply:
Just to be sure that a label whose value is empty is not added to the selector spec.